pkg/git/libgit2/checkout.go

```diff
@@ -618,18 +519,49 @@ func initializeRepoWithRemote(ctx context.Context, path, url string, opts *git.A
 				repo.Free()
 				return nil, nil, fmt.Errorf("unable to create or lookup remote '%s'", defaultRemoteName)
 			}
-			if remote.Url() != url {
-				repo.Free()
-				return nil, nil, fmt.Errorf("remote '%s' with different address '%s' already exists", defaultRemoteName, remote.Url())
+
+			if remote.Url() != transportOptsURL {
+				err = repo.Remotes.SetUrl("origin", transportOptsURL)
+				if err != nil {
+					repo.Free()
+					remote.Free()
+					return nil, nil, fmt.Errorf("unable to configure remote %s origin with url %s", defaultRemoteName, url)
+				}
+
+				// refresh the remote
+				remote, err = repo.Remotes.Lookup(defaultRemoteName)
+				if err != nil {
+					repo.Free()
+					return nil, nil, fmt.Errorf("unable to create or lookup remote '%s'", defaultRemoteName)
+				}
```

Comment:
This too looks like a significant change in the behavior mixed with other changes.
It'd be good to move this too into it's own commit with it's story and highlight this change in the PR description. It'll make it easier to track the change if it cause any problem later.
I see that the function description talks about it. Would be good to keep that and this block of change together.